diff --git a/contracts/token/ERC721/ERC721Upgradeable.sol b/contracts/token/ERC721/ERC721Upgradeable.sol
index 17d24dbd..6e903be8 100644
--- a/contracts/token/ERC721/ERC721Upgradeable.sol
+++ b/contracts/token/ERC721/ERC721Upgradeable.sol
@@ -135,7 +135,7 @@ abstract contract ERC721Upgradeable is Initializable, ContextUpgradeable, ERC165
     }
 
     /// @inheritdoc IERC721
-    function transferFrom(address from, address to, uint256 tokenId) public virtual {
+    function transferFrom(address from, address to, uint256 tokenId) public virtual payable {
         if (to == address(0)) {
             revert ERC721InvalidReceiver(address(0));
         }
@@ -148,12 +148,12 @@ abstract contract ERC721Upgradeable is Initializable, ContextUpgradeable, ERC165
     }
 
     /// @inheritdoc IERC721
-    function safeTransferFrom(address from, address to, uint256 tokenId) public {
+    function safeTransferFrom(address from, address to, uint256 tokenId) public virtual payable {
         safeTransferFrom(from, to, tokenId, "");
     }
 
     /// @inheritdoc IERC721
-    function safeTransferFrom(address from, address to, uint256 tokenId, bytes memory data) public virtual {
+    function safeTransferFrom(address from, address to, uint256 tokenId, bytes memory data) public virtual payable {
         transferFrom(from, to, tokenId);
         ERC721Utils.checkOnERC721Received(_msgSender(), from, to, tokenId, data);
     }
